contracts/peripheral/AuraViewHelpers.sol

```diff
@@ -0,0 +1,441 @@
+        uint256 withdrawalPenalty;
+        ExtraRewards[] extraRewards;
+    }
+
+    struct VaultAccount {
+        address addr;
+        uint256 balance;
+        uint256 balanceOfUnderlying;
+        uint256[] extraRewardsEarned;
+    }
+
+    struct Locker {
+        uint256 epoch;
+        uint256 totalSupply;
+        uint256 lockedSupply;
+        RewardsData rewardsData;
+    }
+
+    struct LockerAccount {
+        address addr;
+        uint256 total;
+        uint256 unlockable;
+        uint256 locked;
+        uint256 nextUnlockIndex;
+        uint128 rewardPerTokenPaid;
+        uint128 rewards;
+        address delegate;
+        uint256 votes;
+        AuraLocker.LockedBalance[] lockData;
+        AuraLocker.EarnedData[] claimableRewards;
+    }
+
+    struct RewardsData {
+        uint256 periodFinish;
+        uint256 lastUpdateTime;
+        uint256 rewardRate;
+        uint256 rewardPerTokenStored;
+        uint256 queuedRewards;
+    }
+
+    struct ExtraRewards {
+        address addr;
+        address rewardsToken;
+        RewardsData rewardsData;
+    }
+
+    struct PoolBalances {
+        uint256 pid;
+        uint256 earned;
+        uint256[] extraRewardsEarned;
+        uint256 staked;
+    }
+
+    function getVault(address _vault) external view returns (Vault memory vault) {
+        IAuraBalVault auraBalVault = IAuraBalVault(_vault);
+
+        address underlying = auraBalVault.underlying();
+        uint256 totalUnderlying = auraBalVault.totalUnderlying();
+        uint256 totalSupply = auraBalVault.totalSupply();
+        uint256 withdrawPenalty = auraBalVault.withdrawalPenalty();
+
+        ExtraRewards[] memory extraRewards = getExtraRewards(_vault);
+
+        vault = Vault({
+            addr: _vault,
+            underlying: underlying,
+            totalUnderlying: totalUnderlying,
+            totalSupply: totalSupply,
+            withdrawalPenalty: withdrawPenalty,
+            extraRewards: extraRewards
+        });
+    }
+
+    function getVaultAccount(address _vault, address _account)
+        external
+        view
+        returns (VaultAccount memory vaultAccount)
+    {
+        IAuraBalVault auraBalVault = IAuraBalVault(_vault);
+
+        uint256 balance = auraBalVault.balanceOf(_account);
+        uint256 balanceOfUnderlying = auraBalVault.balanceOfUnderlying(_account);
+
+        uint256 extraRewardsLength = auraBalVault.extraRewardsLength();
+        uint256[] memory extraRewardsEarned = new uint256[](extraRewardsLength);
+        for (uint256 i = 0; i < extraRewardsLength; i++) {
+            IBaseRewardPool extraRewardsPool = IBaseRewardPool(auraBalVault.extraRewards(i));
+            extraRewardsEarned[i] = extraRewardsPool.earned(_account);
+        }
+
+        vaultAccount = VaultAccount({
+            addr: _account,
+            balance: balance,
+            balanceOfUnderlying: balanceOfUnderlying,
+            extraRewardsEarned: extraRewardsEarned
```

Comment:
There is a library that calculates  how much aura will be mined based on the earned rewards, you might be interested at  `contracts/peripheral/AuraMining.sol`

Reply:
True, this probably could be done here; currently I'm offloading this job to the consumer of this contract. Only reason I can think not to do this is that it might not be as portable across other chains.